bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttService.java

```diff
@@ -29,211 +34,315 @@
-         */
-        for (final MqttBrokerConnection conn : brokerConnections.values()) {
-            try {
-                conn.start();
-            } catch (final Exception e) {
-                conn.connectionLost(e);
-                logger.error("Error starting broker connection", e);
+    /**
+     * Add a listener to get notified of new/removed brokers.
+     *
+     * @param observer The observer
+     */
+    public void addBrokersListener(MqttBrokersObserver observer) {
+        brokersObservers.add(observer);
+    }
+
+    /**
+     * Remove a listener and don't get notified of new/removed brokers anymore.
+     *
+     * @param observer The observer
+     */
+    public void removeBrokersListener(MqttBrokersObserver observer) {
+        brokersObservers.remove(observer);
+    }
+
+    /**
+     * Lookup an broker connection by name.
+     *
+     * @param brokerName to look for.
+     * @return existing connection or null
+     */
+    public MqttBrokerConnection getBrokerConnection(String brokerName) {
+        synchronized (brokerConnections) {
+            return brokerConnections.get(brokerName.toLowerCase());
+        }
+    }
+
+    /**
+     *
+     * @param connection
+     */
+    public void addBrokerConnection(MqttBrokerConnection connection) {
+        synchronized (brokerConnections) {
+            brokerConnections.put(connection.getName(), connection);
+            for (MqttBrokersObserver o : brokersObservers) {
+                o.brokerAdded(connection);
             }
         }
     }
 
-    private void stopAllBrokerConnections() {
-        /*
-         * No need to synchronize the access to the broker connections.
-         * We don't add or remove entries only use the existing ones and the map is a concurrent one.
-         */
-        for (final MqttBrokerConnection conn : brokerConnections.values()) {
-            logger.info("Stopping broker connection '{}'", conn.getName());
-            conn.close();
+    /**
+     * Add a broker by a configuration key-value map. You need to provide at least a "name" and an "url".
+     * Additional properties are "user","pwd","qos","retain","lwt","keepAlive","clientId", please read the
+     * service configuration documentation for a detailed description.
+     *
+     * @param brokerConnectionConfig The configuration key-value map.
+     * @return Returns the created broker connection.
+     * @throws ConfigurationException Most likely your provided name and url are invalid.
+     * @throws MqttException
+     */
+    public MqttBrokerConnection addBrokerConnection(Map<String, Object> brokerConnectionConfig)
+            throws ConfigurationException, MqttException {
+        String name = brokerConnectionConfig.get("name").toString();
+        if (name == null) {
+            throw new ConfigurationException("MQTT Broker property '{}' should have the format 'broker.propertykey'");
```

Comment:
there is no value for `{}` here. You can name the `name` property right away here.